pkg/smokescreen/config.go

```diff
@@ -87,6 +87,9 @@ type Config struct {
 	// Custom handler to allow clients to modify reject responses
 	RejectResponseHandler func(*http.Response)
 
+	// Custom handler to allow clients to modify reject responses
+	RejectResponseHandlerWithCtx func(*SmokescreenContext, *http.Response)
```

Comment:
Should we guard against configuring both this handler and the legacy one? If not, we'd need to document ordering of handler execution et all for consumers.

Reply:
Yes, it makes sense to only let users configure one of these handler since new one is the superset of legacy one.
I can mark legacy one as depreciated.